src/disasm.rs

```diff
@@ -0,0 +1,524 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Access to the Disasm struct.
+//!
+//! This module contains the Disasm struct which can be used to parse the output file of a objdump command.
+//! This file operates over files generated with the following combination of flags:
+//! objdump -d --no-addresses --no-show-raw-insn
+use std::fmt;
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::BufReader;
+
+mod instruction;
+mod section;
+mod symbol;
+
+use instruction::Instruction;
+use section::Section;
+use symbol::Symbol;
+
+use lazy_static::lazy_static;
+use regex::Regex;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Disasm {
+    _file_name: String,
+    _file_format: String,
+    sections: Vec<Section>,
+}
+
+impl Disasm {
+    fn from_lines(lines: Vec<String>) -> Result<Self, String> {
+        let mut disasm = Disasm {
+            _file_name: String::from(""),
+            _file_format: String::from(""),
+            sections: Vec::new(),
+        };
+        // Filter out empty lines
+        let mut lines_iter = lines.into_iter().filter(|line| !line.trim().is_empty());
+        // Process the first line
+        let line = lines_iter
+            .next()
+            .ok_or("Error, the file does not contain any text".to_string())?;
+        disasm.process_first_line(line)?;
+        // Process all other lines
+        for line in lines_iter {
+            disasm.process_other_line(line)?;
+        }
+        // Sort the stored data
+        disasm.sort_sections();
+        Ok(disasm)
+    }
+
+    fn process_first_line(&mut self, line: String) -> Result<(), String> {
+        let err_msg = "Incorrect format for the first line";
+
+        let (file_name, leftover_line) = line.split_once(':').ok_or(err_msg.to_string())?;
+        self._file_name = file_name.to_string();
+        self._file_format = leftover_line
+            .trim()
+            .strip_prefix("file format ")
+            .ok_or(err_msg.to_string())?
+            .to_string();
+        Ok(())
+    }
+
+    fn process_other_line(&mut self, line: String) -> Result<(), String> {
+        lazy_static! {
+            static ref RE_SECTION: Regex =
+                Regex::new(r"^Disassembly of section (?P<sec_name>.[[:alnum:].]+):$").unwrap();
+            static ref RE_SYMBOL: Regex = Regex::new(r"^(?P<sym_name><.+>):$").unwrap();
+            static ref RE_INSTRUCTION: Regex = Regex::new(
+                r"(?x)^
+                    [[:space:]]
+                    (?P<opcode>  [[[:lower:]][[:digit:]][[:space:]]]*)
+                    (?P<operands>[[:space:]]+[^[[:space:]]]+)??
+                    ([[:space:]]+\#(?P<comment>.*))??
+                    [[:space:]]*
+                    $"
+            )
+            .unwrap();
+        }
```

Comment:
I looked into `nom` today and yesterday. I agree that it's a very nice approach.
However, the learning curve of a tool like `nom` is very steep in Rust. Primarily because of the rigid types in Rust.
I don't think it makes sense to include those changes in this PR, as it would take me days to figure `nom` out.
Also, the Excercise 2+ changes quite a lot the regexes, so better to move to a parser combinator later on.
I will attempt to open an issue about it :)